### PR TITLE
Add undefine query

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/UndefineQueryImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import javax.annotation.Nullable;
+import java.util.stream.Stream;
+
+import static ai.grakn.util.CommonUtil.toImmutableList;
+
+/**
+ * @author Felix Chapman
+ */
+@AutoValue
+abstract class UndefineQueryImpl implements UndefineQuery {
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/UndefineQueryImpl.java]: What's with all the nulls?

File: grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java [hunk 2]
```diff
@@ -308,10 +308,8 @@ Shard buildShard(Vertex vertex){
 
     @Nullable
     public VertexElement buildVertexElement(Vertex vertex){
-        try {
-            graknGraph.validVertex(vertex);
-        } catch (IllegalStateException e){
-            LOG.warn("Invalid vertex [" + vertex + "] due to " + e.getMessage(), e);
+        if (!graknGraph.validElement(vertex)) {
```
Comment (fppt) [inline grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java]: What's a `graknGraph` ?

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/UndefineQueryImpl.java (hunk 1 above)
Comment (aelred) [reply grakn-graql/src/main/java/ai/grakn/graql/internal/query/UndefineQueryImpl.java]: haha

it's been a long day

will fix

File: grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java (hunk 2 above)
Comment (aelred) [reply grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java]: A `graknGraph` is a nearly perfect sphere of hot plasma. It is by far the most important source of energy for life on Earth. Its diameter is about 109 times that of Earth, and its mass is about 330,000 times that of Earth, accounting for about 99.86% of the total mass of the Solar System. About three quarters of a `graknGraph`'s mass consists of hydrogen (~73%); the rest is mostly helium (~25%), with much smaller quantities of heavier elements, including oxygen, carbon, neon, and iron.

Comment (fppt) [reply grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java]: Ah thank you for clarifying that!

**Pull request merged**